src/main/java/com/google/sps/servlets/GetTAWorkspace.java

```diff
@@ -0,0 +1,78 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.UserRecord;
+import com.google.gson.Gson;
+import com.google.sps.firebase.FirebaseAppManager;
+import java.io.IOException;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-TA-workspace")
+public class GetTAWorkspace extends HttpServlet {
```

Comment:
Could this servlet be combined with `get-student-workspace` so that there is only one servlet, `get-workspace`? It looks like both of these servlets do essentially the same thing except for how they handle authentication. Maybe the authentication for the combined servlet could be to check whether the user is the student OR the TA.